docs/api/navigators/StackNavigator.md

```diff
@@ -118,7 +118,7 @@ All `navigationOptions` for the `StackNavigator`:
 
 - `title` - a title (string) of the scene
 - `header` - a config object for the header bar:
-  - `visible` - Boolean toggle of header visibility. Only works when `headerMode` is `screen`.
+  - `bar` - React Element to display as a header or `null` to hide it. Only works when `headerMode` is `screen`
```

Comment:
> What happens when there's nothing to go back to?

You have an access to `navigation` so you can read the state and decide what's the index. It's not different from what we currently do. 

> Also how'd you render something like previous screen title in back button

Same as above? I think in that case, I think it could be either a function that receives `{...transitionProps, router}` (just like header) or an element for those who need a simple use-case.